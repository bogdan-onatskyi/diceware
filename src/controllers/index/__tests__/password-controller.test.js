import React from 'react';

import PasswordController from '../password-controller';
import passwordObject from '../../../models/password-object';

describe("PasswordController component", () => {
    it("should match saved snapshot (passwordObject.isVariantsOpened === false)", () => {
        passwordObject.wordArray.forEach((wordObject) => {
            wordObject.index = 0;
        });

        const wrapper = shallow(
            <PasswordController passwordObject={passwordObject}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should match saved snapshot (passwordObject.isVariantsOpened === true)", () => {
        passwordObject.wordArray.forEach((wordObject) => {
            wordObject.index = 0;
        });
        passwordObject.isVariantsOpened = true;

        const wrapper = shallow(
            <PasswordController passwordObject={passwordObject}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should have 1 prop", () => {
        const propsStr = 'passwordObject';
        const props = [];
        for (let prop in PasswordController.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});