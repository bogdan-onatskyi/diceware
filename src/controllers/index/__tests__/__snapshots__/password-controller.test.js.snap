// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordController component should match saved snapshot (passwordObject.isVariantsOpened === false) 1`] = `
<div
  className="password-view"
>
  <Password
    isVariantsOpened={false}
    password="ABACUS ABACUS ABACUS ABACUS ABACUS"
    toggleVariantsOpened={[Function]}
  />
</div>
`;

exports[`PasswordController component should match saved snapshot (passwordObject.isVariantsOpened === true) 1`] = `
<div
  className="password-view"
>
  <Password
    isVariantsOpened={true}
    password="ABACUS ABACUS ABACUS ABACUS ABACUS"
    toggleVariantsOpened={[Function]}
  />
  <Variants
    caps={63}
    isCAPS={[Function]}
    separatedPassword={[Function]}
    separators={
      Array [
        "-",
        "_",
        ":",
        ".",
        "",
        " ",
      ]
    }
    toggleCAPS={[Function]}
  />
</div>
`;
